### ISSUE TITLE
Move candidate pagination to backend

### ISSUE THREAD

**Issue by Safi-Ullah** [fenske/devskills-app#216]
Even though the candidate listing page is paginated but the app currently fetches the full candidate list from the server which is not an ideal solution.

## Acceptance Criteria
- Move pagination to backend i.e retrieve only the relevant paginated records from the server.
- Set page size to 10.

**Comment by Safi-Ullah** [comment]
## Grooming

- Need to move current filters/sorting to the backend as well.
- Pass already formed queryParams to the API as-is.
- Modify the query within the following function to run on the params passed. 

<img width="1440" alt="Screenshot 2021-05-05 at 4 31 43 AM" src="https://user-images.githubusercontent.com/29250597/117082106-cc3ace80-ad5a-11eb-8877-f95e5e5a8fd6.png">

**Comment by fenske** [comment]
Thanks for creating this issue. I've removed the 'bug' label as it rather felt more like an enhancement. :)

**Comment by Safi-Ullah** [comment]
> Thanks for creating this issue. I've removed the 'bug' label as it rather felt more like an enhancement. :)

Yeah, I wanted to remove it but seems like I can't modify(add/remove) them.
